src/uct/ib/rc/base/rc_ep.h

```diff
@@ -248,8 +254,14 @@ void uct_rc_fc_cleanup(uct_rc_fc_t *fc);
 
 ucs_status_t uct_rc_ep_fc_grant(uct_pending_req_t *self);
 
+void uct_rc_txqp_purge_outstanding_cb(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp,
```

Comment:
why need to change this semantic?

Reply:
because DCX failure may require own logic: re-post requests 2-N

Reply:
can't you just invoke different func for keepalive DCI?

Reply:
I can, and in arbiter as well, but it will produce code duplication

Reply:
reverted purge_outstanding changes